src/averageAges.js

```diff
@@ -54,10 +49,54 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
+function calculateMenAverageAge(people, century) {
+  const men = century
+    ? people
+      .filter(person => Math.ceil(person.died / 100) === century)
+      .filter(findMen)
+    : people.filter(findMen);
+
+  return calculate(men);
+}
+
+function calculateWomenAverageAge(people, withChildren) {
+  const women = withChildren
+    ? people
+      .filter(findWomen)
+      .filter(person => people.find(mother => mother.mother === person.name))
```

Comment:
it's better to find some other name because mother.mother looks weird and not very obvious😅